### DISCUSSION TITLE
Adding an API to enable/disable parents and routes

### DISCUSSION THREAD

**Discussion by mathpi314159-fap** [ozkeisar/mockingbird#32] [Ideas]
In order to facilitate the automation of end-to-end testing, I propose the addition of API methods for enabling/disabling parent and routing. I would like to run the server for the end-to-end (E2E) test suite. However, not all tests require a server for moking requests. I would appreciate guidance on how to enable the parent and route options.

**Comment by ozkeisar** [comment]
If I understand correctly you want an API to enable/disable "block proxy" on spesific routes?

**Reply by mathpi314159-fap** [reply]
Yes, the preset mechanism seems to be close to what I require. However, there is no option to disable it.

For instance, I wish to block the /api/v1/config endpoint. I will accomplish this using the present method.
Using the framework for end-to-end (E2E) testing, I will invoke the API request: /presets/{projectName}/{presetFolderName}/{presetName}/apply.
However, how can I cancel a preset? For other tests, I require an unmodified /api/v1/config request.

**Reply by mathpi314159-fap** [reply]
Yes, I need an API to enable/disable "proxy blocking" on certain routes

**Reply by ozkeisar** [reply]
It sounds like all you need is to create new preset that will set all you routes back to the "default" state that you want, no?

**Reply by mathpi314159-fap** [reply]
Yes, this method would be suitable for my needs. I have attempted to implement it. I will need to add default routes to the preset. The issue here is distinct. And, how do I ensure that a route returns a request response without any changes?

**Reply by ozkeisar** [reply]
Add a default response to your routes, and then create preset that sets the default response for all your route to be the active response.
That will answer your needs
